iceoryx_hoofs/test/moduletests/test_cxx_variant.cpp

```diff
@@ -128,26 +129,48 @@ bool variant_Test::DTorTest::dtorWasCalled = false;
 int variant_Test::DoubleDelete::dtorCalls = 0;
 int variant_Test::DoubleDelete::ctorCalls = 0;
 
-TEST_F(variant_Test, DefaultCTor)
+TEST_F(variant_Test, DefaultCTorCreatesInvalidVariant)
 {
     EXPECT_THAT(sut.index(), Eq(iox::cxx::INVALID_VARIANT_INDEX));
 }
 
-TEST_F(variant_Test, emplaceValidElement)
+TEST_F(variant_Test, InitializedVariantReturnsCorrectIndex)
 {
-    ASSERT_THAT(sut.emplace<ComplexClass>(123, 456.789f), Eq(true));
+    sut.emplace<float>(1231.22F);
+    EXPECT_THAT(sut.index(), Eq(1U));
+}
+
+TEST_F(variant_Test, CreatingVariantFromPlainTypeReturnsProvidedValueAndCorrectIndex)
+{
+    iox::cxx::variant<ComplexClass, float> sut2{42.42F};
+
+    EXPECT_THAT(*sut2.get<float>(), Eq(42.42F));
+    EXPECT_THAT(sut2.index(), Eq(1U));
+}
+
+TEST_F(variant_Test, CreatingVariantFromLValueReturnsProvidedValueAndCorrectIndex)
+{
+    std::string string("Buhh");
+    iox::cxx::variant<std::string, float> sut2{string};
+    EXPECT_THAT(sut2.get<std::string>()->c_str(), StrEq("Buhh"));
+    EXPECT_THAT(sut2.index(), Eq(0U));
+}
+
```

Comment:
Could you please add a test for `cxx::expected<float, float>`?

Reply:
I'm sure you meant `variant<float, float>` ;) Added.